File: backend/app/schemas/quality.py
from pydantic import BaseModel


class DefectCreate(BaseModel):
    project_id: str
    title: str
    description: str
    severity: str
    priority: str
    related_test_result_id: str | None = None
    related_test_case_id: str | None = None


class DefectOut(BaseModel):
    id: str
    project_id: str
    title: str
    severity: str
    priority: str
    status: str


class QualityScoreOut(BaseModel):
    overall: int
    functional: int
    reliability: int
    security: int
    performance: int
    accessibility: int
    coverage: int


class ReleaseReadinessOut(BaseModel):
    status: str
    pass_rate: float
    critical_defects: int
    quality_score: int
    gate_violations: list[str]

File: frontend/src/services/api/quality.ts
import { apiClient } from "./client";

export interface QualityScoreOut {
  overall: number;
  functional: number;
  reliability: number;
  security: number;
  performance: number;
  accessibility: number;
  coverage: number;
}

export interface ReleaseReadinessOut {
  status: string;
  pass_rate: number;
  critical_defects: number;
  quality_score: number;
  gate_violations: string[];
}

export async function getQualityScore(projectId: string): Promise<QualityScoreOut> {
  const { data } = await apiClient.get<QualityScoreOut>("/api/quality/score", { params: { project_id: projectId } });
  return data;
}

export async function getReleaseReadiness(projectId: string): Promise<ReleaseReadinessOut> {
  const { data } = await apiClient.get<ReleaseReadinessOut>("/api/quality/release-readiness", { params: { project_id: projectId } });
  return data;
}
